src/protocol/profiler.rs

```diff
@@ -0,0 +1,87 @@
+use serde::Deserialize;
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+/// Coverage data for a source range.
+pub struct CoverageRange {
+    pub start_offset: u16,
+    pub end_offset: u16,
+    pub count: u16,
```

Comment:
The original types for these are `integer`, for which we better use `u64` to prevent overflow, especially on the counter.

Reply:
So what they call integer is actually an i32, and if you go over that you get an error. I figured for non negative numbers a u16 makes just as much sense, but perhaps it'd be better to just use i32s everywhere when dealing with the protocol, just in case.

Reply:
I've used u64 elsewhere before. Maybe a type alias `JsInt = i32` and `JsUInt = u32` so those members read `pub count: JsUInt` and make clear why this particular type

Reply:
The thing with "`i32` everywhere" is that one is semi-implicitly in signed-arithmetic-land then, which does not make sense e.g. for a counter.

Reply:
You only get the error of you go over the max value of i32

Reply:
I was thinking more about going to -1 instead of getting an underflow abort

Reply:
Ah sorry, that comment was in response to your comment about having used u64. Agreed re type aliases, except that using u32 for unsigned JS int would allow for sending numbers too large

Reply:
That is also true but without resorting to special integer types one thing has to give :)

Reply:
True! I'll make that change.